### DISCUSSION TITLE
Allow cookie name for overriding accept-language header to be configurable

### DISCUSSION THREAD

**Discussion by jcolakk** [vercel/next.js#24852] [Ideas]
### Describe the feature you'd like to request

Currently, for i18n, Nextjs only allow `NEXT_LOCALE` [as a name](https://nextjs.org/docs/advanced-features/i18n-routing#leveraging-the-next_locale-cookie) for a cookie containing the locale.
It would be great if this would be configurable through next.config.js since I work in several apps in which we leverage the cookie already for locale and it is named differently. 

### Describe the solution you'd like

Allow cookie configuration in next.config.js:

```
module.exports = {
   i18n: {
        locales: Object.keys(languages),
        defaultLocale: 'en',
        cookieName: 'preferred_language'
     },
}
```

### Describe alternatives you've considered

I don't have any alternatives. I tried hacking it to keep the `NEXT_LOCALE` and my own locale cookie in sync, but its really brittle solution.

**Comment by akafaneh** [comment]
Yes please we are facing a similar issue
